From david.vrabel@csr.com  Thu Jun 19 11:53:10 2008
From: David Vrabel <david.vrabel@csr.com>
Date: Mon, 09 Jun 2008 16:01:45 +0100
Subject: UWB: fix oops in uwbd_dev_offair()
To: linux-usb@vger.kernel.org
Cc: Greg KH <greg@kroah.com>
Message-ID: <484D45D9.9070307@csr.com>


From: David Vrabel <david.vrabel@csr.com>

If a radio controller sees only 1 or two beacons from a device, then
there is a beacon cache entry without a uwb_dev that will (eventually)
be a candidate for purging.

Signed-off-by: David Vrabel <david.vrabel@csr.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/uwb/lc-dev.c |    9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

--- a/drivers/uwb/lc-dev.c
+++ b/drivers/uwb/lc-dev.c
@@ -437,14 +437,13 @@ int __uwb_dev_offair(struct uwb_dev *uwb
  */
 void uwbd_dev_offair(struct uwb_beca_e *bce)
 {
-	struct uwb_rc *rc;
 	struct uwb_dev *uwb_dev;
 
 	uwb_dev = bce->uwb_dev;
-	rc = uwb_dev->rc;
-
-	uwb_notify(rc, uwb_dev, UWB_NOTIF_OFFAIR);
-	__uwb_dev_offair(uwb_dev, rc);
+	if (uwb_dev) {
+		uwb_notify(uwb_dev->rc, uwb_dev, UWB_NOTIF_OFFAIR);
+		__uwb_dev_offair(uwb_dev, uwb_dev->rc);
+	}
 }
 
 
